index.js

```diff
@@ -189,19 +200,32 @@ Metric.prototype.put = function(value, metricName, additionalDimensions) {
  * Metric instance to track those two summary sets independently!
  * @param {Number} value The value to include in the summary.
  * @param {String} metricName The name of the metric we're summarizing.
+ * @param {String} units CloudWatch units
  * @param {Object[]} additionalDimensions The extra dimensions we're tracking.
  */
-Metric.prototype.summaryPut = function(value, metricName, additionalDimensions = []) {
-  const key = makeKey(metricName, additionalDimensions);
+Metric.prototype.summaryPut = function(...args) {
+  if (args.length === 3) {
+    const [value, metricName] = args;
+    const units = Array.isArray(args[2]) ? this.units : args[2];
```

Comment:
Let's do the same as above and extract the `Array.isArray(args[2])` check so that we can reuse it here and the line below.